공통 컴포넌트 개발

minjyo opened the pull request.

minjyo committed:
chore: gitignore 수정

ummaeha committed:
feat:: 공통 버튼 컴포넌트 구현

minjyo committed:
feat: Header 컴포넌트, Layout 추가 및 styled-component 설정

minjyo committed:
fix: resolve merge conflict

minjyo committed:
refactor: Header.tsx, Layout.tsx 를 components/common 폴더로 이동

minjyo committed:
refactor: 디렉토리 구조 수정

minjyo committed:
refactor: theme에서 breakpoints와dark/light color 추가

minjyo committed:
feat: DefaultButton, FilledButton 컴포넌트 추가

minjyo committed:
feat: SubTitle 컴포넌트 추가

minjyo committed:
feat: IconBox 컴포넌트 추가

minjyo committed:
feat: MenuBox 컴포넌트 추가

minjyo committed:
chore: MenuBox 컴포넌트 스타일 변경 및 component selector 추가

minjyo committed:
feat: Profile 컴포넌트 추가

minjyo committed:
fix: Profile 컴포넌트 props 수정

minjyo committed:
feat: Dot 컴포넌트 추가

minjyo committed:
feat: NavBar, FriendBox 컴포넌트 추가

minjyo committed:
feat: Text 컴포넌트 추가

minjyo committed:
chore: Layout 구조 변경

ummaeha committed:
feat:  header 컴포넌트 추가

ummaeha committed:
chore: pull 이후 컨플릭트 수정

ummaeha committed:
refactor: Layout.tsx 구조 변경

ummaeha committed:
refactor: Text 컴포넌트 types props 디폴트 값을 main으로  지정

ummaeha committed:
feat: json-server 설정, db.json 파일 추가

ummaeha commented:
https://github.com/STOVE-Milk/steam-clone/pull/3/commits/c104ff1f4a9d7d33017dc6255eccd92c0886dfaa
json-server는 해당 디렉토리(/db)에 진입해서 `npm start`로 실행하시면 됩니다. 
4001번 포트를 사용하고있고, API 명세 파일 참고하여 `db.json`파일에 데이터베이스를 작성하면 됩니다.
(일단 서버개발자분꼐 mock 데이터까지 요청드린 상태)

minjyo committed:
feat: carousel 컴포넌트 추가

minjyo committed:
fix: merge conflict resolve

ummaeha committed:
chore: axios 셋팅 추가

minjyo committed:
refactor: carousel 컴포넌트 organism으로 분리

minjyo committed:
fix: merge conflict 해결

minjyo committed:
feat: GameSlide 컴포넌트 추가

minjyo committed:
feat: Carousel 컴포넌트 props 수정

minjyo committed:
chore: GameSlide 컴포넌트 css 수정

minjyo committed:
chore: Carousel 컴포넌트 css 수정

ummaeha committed:
refactor: header컴포넌트 머지 전 수정
1. searchbox radius 10px 통일
2. <select> 부분 molecule로 빼서 컴포넌트화하기
3. <select>에서 <option> 선택했을 때 상태 처리(setState) → 아직 검색 옵션$
4. 디자인 통일 -> 배경색 x , 버튼 -> icon, 알림 아이콘 유저아이콘 위치

ummaeha committed:
Merge branch 'feature/front-common' of https://github.com/STOVE-Milk/steam-clone into feature/front-common

minjyo committed:
chore: Carousel 컴포넌트 css 수정

minjyo committed:
Merge branch 'feature/front-common' of github.com:STOVE-Milk/steam-clone into feature/front-common

ummaeha commented on src/frontend/components/molecules/FriendBox.tsx:
```diff
@@ -0,0 +1,48 @@
+const FriendName = styled(Text)`
+  margin-left: 20px;
+  margin-top: 5px;
+  flex: 1;
```
저는 flex:1 을 잘 사용하지 않았는데, 지호님 덕분에 알아갑니다 :)

ummaeha commented on src/frontend/components/molecules/GameSlide.tsx:
```diff
@@ -0,0 +1,52 @@
+}
+
+const SlideWrapper = styled.div`
+  width: 200px;
```
px로 고정되는 부분에 대해 이야기 나눠봐야할 것 같아요. 반응형 제작하기 시작하면 px 고정이 방해될 수 있을 것 같습니다.

minjyo replied:
네 이 부분은 Carousel 라이브러리 내에서 Item을 반응형 처리를 자동으로 해주어서, width: 100%;로 변경했습니다!

ummaeha commented on src/frontend/components/molecules/GameSlide.tsx:
```diff
@@ -0,0 +1,52 @@
+      <ImageSection>{props.image}</ImageSection>
+      <InfoSection>
+        <Info types="small">{props.info.name}</Info>
+        <Info types="tiny">{`$ ${props.info.price}`}</Info>
```
javascript의 toLocaleString() 메서드를 사용해보세요. 돈 단위와 세자리 단위에 따른 쉼표처리를 해줍니다.
필요에따라 함수화 해서 여러군데서 씌워서 사용해도 될 것 같습니다.

https://www.w3schools.com/jsref/jsref_tolocalestring_number.asp

minjyo replied:
오 이런게 있네요! 
저희 앞으로 게임 가격 관련해서 많이 쓰일거같으니 util 안에 만들어두겠습니다!

ummaeha commented on src/frontend/components/molecules/SearchBox.tsx:
```diff
@@ -0,0 +1,67 @@
+import React, { useState, useCallback, useMemo } from 'react';
```
제껀데... useMemo안쓰니 지우도록 하겠습니다. @ummaeha

ummaeha commented on src/frontend/components/organisms/Carousel.tsx:
```diff
@@ -0,0 +1,60 @@
+    background: ${(props) => props.theme.colors.secondaryBg};
+  }
+  .react-multiple-carousel__arrow--left {
+    margin-left: -3%;
```
마이너스 마진으로 위치조절을 하다니 독특하네요.!

ummaeha commented on src/frontend/pages/_app.tsx:
```diff
@@ -1,16 +1,19 @@
     <ThemeProvider theme={theme}>
-        <GlobalStyle />
+      <GlobalStyle />
+      <Layout>
```
👍

ummaeha commented on src/frontend/package.json:
```diff
@@ -8,14 +8,20 @@
     "react-dom": "^17.0.2",
-    "styled-components": "^5.3.3",
-    "styled-normalize": "^8.0.7"
+    "react-konva": "^17.0.2-5",
```
konva 설치를 하셨군요!!!

ummaeha commented on src/frontend/components/organisms/map.tsx:
```diff
@@ -0,0 +1,60 @@
+import { Stage, Layer, Circle } from 'react-konva';
```
konva관련 테스트 파일은 common 브랜치에서 제외해야할 것 같습니다 :)

minjyo replied:
네 지우고 다시 커밋하겠습니다~

minjyo committed:
chore: Carousel 컴포넌트 css 수정

minjyo committed:
feat: global-style에 html의  media query에 font-size 추가

minjyo committed:
chore: react-konva test 관련 파일 삭제

minjyo committed:
fix: key props warning 삭제

minjyo committed:
feat: util/localeString에 localPrice 함수 추가

minjyo commented on src/frontend/components/atoms/DefaultButton.tsx:
```diff
@@ -0,0 +1,37 @@
+  font-size: 1rem;
+  cursor: pointer;
+  margin: 0 1em;
+  padding: 0.25em 1em;
```
이 부분 rem 으로 다시 수정할까요?

minjyo commented on src/frontend/components/atoms/FilledButton.tsx:
```diff
@@ -0,0 +1,41 @@
+  font-size: 1em;
+  cursor: pointer;
+  margin: 0 1em;
+  padding: 0.25em 1em;
```
rem !

minjyo commented on src/frontend/components/molecules/SearchBox.tsx:
```diff
@@ -0,0 +1,67 @@
+
+  const handleInputText = useCallback(() => {
+    const searchText = document.querySelector('#searchText')?.value;
+    setInputText(searchText);
```
저는 이 부분 querySelector보다는 
(e) => {
  const searchText = e.target.value; 
}
로 작성해요. selector를 쓰려면 또 선택자를 찾아봐야하니까..

minjyo commented on src/frontend/components/molecules/SearchBox.tsx:
```diff
@@ -0,0 +1,67 @@
+  cursor: pointer;
+`;
+
+export default function Search({ option, inputText, setOption, setInputText }: ISearchbox) {
```
export default 하는 컴포넌트명이랑 파일이름이랑 통일시켜주세요~!

ummaeha committed:
chore: rm -> rem, searchBox export 이름 수정

ummaeha committed:
chore: merge conflict 처리, 불필요한 테스트용 코드 제거
